src/components/dropdown-menu/DropdownMenuSubMenu.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import styled from '@emotion/styled';
+import * as radixDropdownMenu from '@radix-ui/react-dropdown-menu';
+import { ReactNode, useContext } from 'react';
+import { isDefined } from '~common/helpers/types';
+import { THEME_DATA_ATTRIBUTE, ThemeContext } from '~utils/theme';
+import { IconArrowRight } from '../icons';
+import { DropdownMenuItemBase } from './DropdownMenuItemBase';
+
+export type A11yAttrs = Pick<React.AriaAttributes, 'aria-controls'> & {
+  id?: string;
+};
+
+interface DropdownMenuSubProps extends Omit<radixDropdownMenu.DropdownMenuSubProps, 'children'> {
+  className?: string;
+  id?: string;
+  isDisabled?: boolean;
+  isOpen?: boolean;
+  isOpenOnMount?: boolean;
+  items: ReactNode | undefined;
+  triggerLabel: string;
+  triggerSuffix?: ReactNode;
+}
+
+export function DropdownMenuSubMenu({
```

Comment:
It seems like we are missing the forwardRef?
It might also explain why it didn't work with the asChild.

Reply:
would we need the forwardRef here, though? This component is never going to be used on its own, only as part of DropdownMenu?

Reply:
I would just do it to be safe, it's a public and exported component, I would add the forwardRef.

I think we should have an internal rule for the design system that says: "If it's exported, we forwardRef and props." it's just gonna avoid us future headaches and always having the same discussion.